web/src/common/index.ts

```diff
@@ -51,3 +54,30 @@ export function prettyPrintTime(seconds: number): string {
   let leftover = Math.round(seconds - minutes * 60);
   return `${minutes}m${leftover}s`;
 }
+
+/**
+ * Create a zoom-independent distance.
+ *
+ * This is useful (e.g.) for using symbol layers whose output reflects the size of a physical feature on the map.
+ *
+ * @param mapMeters The length on the map in meters
+ * @param minimumPixels A minimum size (in pixels) at all zoom levels. You can set it to 0 if you want the feature to disappear at low zoom levels.
+ */
+export function mapMetersToPixels(
+  mapMeters: number,
+  minimumPixels: number = 4,
+): DataDrivenPropertyValueSpecification<number> {
+  // Random googling gives me these values - I'm not entirely sure they're correct but they seem about right.
```

Comment:
!!! I've been trying to figure out how to do something like this in maplibre since the beginning!! Do you remember what you searched for to get this to work? I thought these values depended on the latitude? But I just tried in both Bristol and Dallas, and the circle size (when zoomed in close enough to see 30 meters) looks fine in both.

I just tried this same function on line widths, and it appears to work! Seriously, this opens up a whole space of design possibilities. Lack of zoom-independent distances has meant I'm previously turning lines into thick polygons on the backend, just to use polygon layers in maplibre.

Reply:
This wasn't the original link, but it has some of the same numbers:

https://gis.stackexchange.com/a/340056